### PR TITLE
Upgradable MultiaddrConnection

### PR DESCRIPTION
**Pull request by ckousik**
We use the  new Upgrader interface to use in-built encryption and multiplexing of a WebRTC connection.

### CODE REVIEW

File: src/sdp.ts [hunk 1]
```diff
@@ -6,7 +6,7 @@ import { bases } from 'multiformats/basics';
 
 const log = logger('libp2p:webrtc:sdp');
 
-const mbdecoder = (function () {
+export const mbdecoder = (function () {
```
Comment (John-LittleBearLabs) [inline src/sdp.ts:9]: Why's this exported?

File: src/stream.ts [hunk 2]
```diff
@@ -44,7 +45,7 @@ export class WebRTCStream implements Stream {
   metadata: Record<string, any>;
   private readonly channel: RTCDataChannel;
 
-  source: Source<Uint8ArrayList> = pushable();
+  _src: Source<Uint8ArrayList> = pushable();
```
Comment (John-LittleBearLabs) [inline src/stream.ts:48]: Do I misremember "source" being part of the interface or has the interface changed?

File: src/stream.ts [hunk 3]
```diff
@@ -168,6 +179,9 @@ export class WebRTCStream implements Stream {
     this.readClosed = true;
     this.writeClosed = true;
     this.channel.close();
+    if (this.closeCb) {
```
Comment (John-LittleBearLabs) [inline src/stream.ts:182]: I had to think for a second about what cb might stand for. Perhaps the name could be more verbose.

File: src/options.ts [hunk 4]
```diff
@@ -6,5 +6,4 @@ export interface WebRTCListenerOptions extends CreateListenerOptions {
   //   channelOptions?: WebRTCReceiverInit
 }
 
-export interface WebRTCDialOptions extends DialOptions {
-}
+export interface WebRTCDialOptions extends DialOptions {}
```
Comment (John-LittleBearLabs) [inline src/options.ts:9]: Shouldn't there be something here specifying your upgrader as the default one?

**Review by John-LittleBearLabs — APPROVED** [review]

File: src/stream.ts (hunk 2 above)
Comment (ckousik) [reply src/stream.ts:48]: No, we just use a getter/setter pattern for accessing `source`.

**Pull request merged**